ui/src/app/subnets/views/VLANDetails/VLANDetailsHeader/VLANDetailsHeader.tsx

```diff
@@ -0,0 +1,24 @@
+import { useSelector } from "react-redux";
+
+import SectionHeader from "app/base/components/SectionHeader";
+import type { RootState } from "app/store/root/types";
+import vlanSelectors from "app/store/vlan/selectors";
+import type { VLAN, VLANMeta } from "app/store/vlan/types";
+
+type Props = {
+  id: VLAN[VLANMeta.PK];
+};
+
+const VLANDetailsHeader = ({ id }: Props): JSX.Element => {
+  const vlan = useSelector((state: RootState) =>
+    vlanSelectors.getById(state, id)
+  );
+
+  if (!vlan) {
+    return <SectionHeader loading />;
+  }
```

Comment:
As with the previous component, IMO this would make more sense to be placed in the main `VLANDetails` component which would be responsible for displaying various states (not found, loading etc.).

Should make this clearer and easier to reason about.